docs/external-helpers.md

```diff
@@ -0,0 +1,110 @@
+
+## How to Add Externalized Helpers
+
+Here’s the step-by-step process to add your own external helper file to the SDK:
+
+### Step 1: Define Helper Functions
+
+External helper functions should be asynchronous and accept the necessary data passed from ZAP, typically via the api object. These helpers can perform any logic, such as querying ZAP's data or processing it.
+
+Here’s an example of a helper that retrieves and counts available clusters:
+
+```javascript
+/**
+ * Helper function to get the total number of available events.
+ * @param {object} api - The API object.
+ * @returns {number} - The total number of events.
+ */
+async function get_total_events_helper(api) {
+  let events = await api.availableEvents(this)
+  let totalEvents = events.length
+  return totalEvents
+}
+```
+
+### In this example:
+
+- The helper accepts the `api` object.
+- It calls `api.availableClusters(this)` to fetch the list of available clusters.
+- It then returns the total number of clusters by calculating the length of the returned array.
+
+Each helper can interact with the ZAP API to fetch data and process it as needed, based on the context passed in.
+
+### Step 2: Specify Helpers in gen-template.json
+
+The next step is to tell ZAP where to find your external helper functions. This is done by specifying the relative path to each helper file in the gen-template.json configuration file.
+
+For example, your gen-template.json might look like this:
+
+```javascript
+{
+  "helpers": [
+    "path/to/external_helpers/get_total_events_helper.js",
+    "path/to/external_helpers/get_total_attributes_helper.js"
+  ]
+}
+```
+
+The paths should be relative to the location of the gen-template.json file itself. This tells ZAP where to find your helper files when generating the binary.
+
+### Step 3: Register Helpers with ZAP
+
+After defining and specifying your helper files, the next step is to register them with ZAP so that they can be used during binary creation. You’ll need to call helperRegister.registerHelpers() for each helper function.
+
+Here’s an example of how to register your helpers:
+
+```javascript
+function initialize_helpers(helperRegister, context) {
+  // Register the 'get_total_events_helper' function
+  helperRegister.registerHelpers(
+    'get_total_events_helper',
+    get_total_events_helper,
+    context
+  )
+
+  // Register other helpers similarly
+  helperRegister.registerHelpers(
+    'get_total_attributes_helper',
+    get_total_attributes_helper,
+    context
+  )
+}
+```
+
+- The `helperRegister.registerHelpers()` function registers each helper.
+- The first argument is the **name** of the helper.
+- The second argument is the **helper function** you defined.
+- The third argument is the **context object**, which contains the necessary environment or data (such as a database connection) that may be passed to the helper.
+
+### Step 4: Use the Helpers in Your SDK Code
```

Comment:
So after registration, the helper function will be global ready to use in any file without import?